zebra-network/src/peer_set/initialize.rs

```diff
@@ -328,9 +329,18 @@ where
                 // turned into a connection, so add it back:
                 let _ = demand_tx.try_send(());
             }
-            // If we don't match one of these patterns, shutdown.
-            _ => break,
+            // We don't expect to see these patterns during normal operation
+            Left((Left((None, _)), _)) => {
+                unreachable!("demand_rx never fails, because we hold demand_tx");
+            }
+            Left((Right((None, _)), _)) => {
+                unreachable!("crawl_timer never terminates");
+            }
+            Right((None, _)) => {
+                unreachable!(
+                    "handshakes never terminates, because it contains a future that never resolves"
```

Comment:
```suggestion
                    "handshake never terminates, because it contains a future that never resolves"
```

Reply:
Thanks for letting us know about potential typos.

`handshakes` is the name of the variable, it's a list:
```rust
    let mut handshakes = FuturesUnordered::new();
    // <FuturesUnordered as Stream> returns None when empty.
    // Keeping an unresolved future in the pool means the stream
    // never terminates.
    handshakes.push(future::pending().boxed());
```